File: old_latex/tcgmsg.c
/* C TCGMSG example */
int main(int argc, char **argv) {
  tcg_pbegin(argc,argv);    /* start TCGMSG */
  GA_Initialize();          /* start global arrays */
  MA_Init(...);             /* start memory allocator** */
  /* ... */                 /* do work */
  GA_Terminate();           /* tidy up global arrays call */
  tcg_pend();               /* tidy up TCGMSG */
}

File: code/tcgmsg.f
! Fortran TCGMSG example
      program main
      include 'tcgmsg.fh'       ! TCGMSG declarations
      call pbeginf()            ! start TCGMSG
      call ga_initialize()      ! start global arrays
      status = ma_init()        ! start memory allocator**
!     ...                       ! do work
      call ga_terminate()       ! tidy up global arrays call
      call pend()               ! tidy up TCGMSG
      end program               ! exit program
